tests/unit/model-test.js

```diff
@@ -487,6 +487,27 @@ test("currentState is accessible when the record is created", function(assert) {
   });
 });
 
+test("Deleting an invalid newly created record should remove it from the store", function(assert) {
+  var record;
+  run(function() {
+    record = store.createRecord('person', { name: 'pablobm' });
+    // Invalidate the record to put it in the `root.loaded.created.invalid` state
+    store.recordWasInvalid(record._internalModel, { name: 'error' });
+  });
+
+  // Preconditions
+  assert.equal(get(record, 'currentState.stateName'), 'root.loaded.created.invalid',
+                 'records should start in the created.invalid state');
+  assert.equal(get(store.peekAll('person'), 'length'), 1, 'The new person should be in the store');
+
+  run(function() {
+    record.deleteRecord();
+  });
+  
+  assert.equal(get(record, 'currentState.stateName'), 'root.deleted.saved');
+  assert.equal(get(store.peekAll('person'), 'length'), 0, 'The new person should be removed from the store');
+});
+
```

Comment:
Just curious: any specific reason why this is not in [tests/integration/records/delete-record-test.js](https://github.com/emberjs/data/blob/master/tests/integration/records/delete-record-test.js)?